src/Transports.AspNetCore/WebSockets/SubscriptionsTransportWs/SubscriptionServer.cs

```diff
@@ -0,0 +1,250 @@
+
+    /// <summary>
+    /// Returns the user context builder used during connection initialization.
+    /// </summary>
+    protected IUserContextBuilder UserContextBuilder { get; }
+
+    /// <summary>
+    /// Returns the <see cref="IGraphQLSerializer"/> used to deserialize <see cref="OperationMessage"/> payloads.
+    /// </summary>
+    protected IGraphQLSerializer Serializer { get; }
+
+    /// <summary>
+    /// Initailizes a new instance with the specified parameters.
+    /// </summary>
+    /// <param name="sendStream">The WebSockets stream used to send data packets or close the connection.</param>
+    /// <param name="options">Configuration options for this instance.</param>
+    /// <param name="authorizationOptions">Authorization options for this instance.</param>
+    /// <param name="executer">The <see cref="IDocumentExecuter"/> to use to execute GraphQL requests.</param>
+    /// <param name="serializer">The <see cref="IGraphQLSerializer"/> to use to deserialize payloads stored within <see cref="OperationMessage.Payload"/>.</param>
+    /// <param name="serviceScopeFactory">A <see cref="IServiceScopeFactory"/> to create service scopes for execution of GraphQL requests.</param>
+    /// <param name="userContextBuilder">The user context builder used during connection initialization.</param>
+    /// <param name="authenticationService">An optional service to authenticate connections.</param>
+    public SubscriptionServer(
+        IWebSocketConnection sendStream,
+        GraphQLWebSocketOptions options,
+        IAuthorizationOptions authorizationOptions,
+        IDocumentExecuter executer,
+        IGraphQLSerializer serializer,
+        IServiceScopeFactory serviceScopeFactory,
+        IUserContextBuilder userContextBuilder,
+        IWebSocketAuthenticationService? authenticationService = null)
+        : base(sendStream, options, authorizationOptions)
+    {
+        DocumentExecuter = executer ?? throw new ArgumentNullException(nameof(executer));
+        ServiceScopeFactory = serviceScopeFactory ?? throw new ArgumentNullException(nameof(serviceScopeFactory));
+        UserContextBuilder = userContextBuilder ?? throw new ArgumentNullException(nameof(userContextBuilder));
+        Serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
+        _authenticationService = authenticationService;
+    }
+
+    /// <inheritdoc/>
+    public override async Task OnMessageReceivedAsync(OperationMessage message)
+    {
+        if (message.Type == MessageType.GQL_CONNECTION_TERMINATE)
+        {
+            await OnCloseConnectionAsync();
+            return;
+        }
+        else if (message.Type == MessageType.GQL_CONNECTION_INIT)
+        {
+            if (Initialized)
+            {
+                await ErrorTooManyInitializationRequestsAsync(message);
+            }
+            else
+            {
+                await OnConnectionInitAsync(message, false);
```

Comment:
🤔 What if 2 `init` requests come concurrently? At least I suggest to test `TryInitialize` result in `OnConnectionInitAsync` to not start timers twice.